src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/BlockDataExtractor.java

```diff
@@ -33,15 +34,18 @@ public BlockDataExtractor(String[] args) throws IOException {
         if (!outputdirectory.exists()) {
             outputdirectory.mkdir();
         }
-        this.outputfile = new File(OUTPUTDIRECTORYNAME + args[1] + ".json");
+        filename = args[1];
+        this.outputfile = new File(OUTPUTDIRECTORYNAME + filename + ".json");
         this.start = Integer.parseInt(args[2]);
         this.stop = Integer.parseInt(args[3]);
 
     }
 
     public void generateJsonFile() throws IOException {
-
-        FileWriter fileWriter = new FileWriter(outputfile, true);
+        if(outputfile.delete()) {
+            outputfile = new File(OUTPUTDIRECTORYNAME + filename + ".json");
+        }
```

Comment:
```suggestion
        outputfile.delete();
```
I don't get why you even need this if statement.

Reply:
mh, i thought it only makes sense to recreate the file, if it is deleted beforehand.

Reply:
What does recreating a file mean in this context?
You are just creating a `File` object in Java here, has nothing to do with what will happen on the actual file system. And the `File` object referenced by `outpufile` has already been created before, with the same arguments.

Reply:
I tested it. you are right. but I still dont get it. at what point is the actual file recreated?

Reply:
Basically here:
https://github.com/internet-sicherheit/ethereum-cache-creator/blob/f85747d7717a3ebd0debed7fcb78dbe43faf4de6/src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/BlockDataExtractor.java#L64